hs/app/reach/Main.hs

```diff
@@ -49,56 +61,30 @@ reachImages =
   ]
 
 
---------------------------------------------------------------------------------
-printfl :: PrintfArg r => String -> r -> TL.Text
-printfl s r = TL.pack $ printf s r
-
-
---------------------------------------------------------------------------------
-echo :: CmdParams p => p
-echo = cmd "echo"
-
-
-sed :: CmdParams p => p
-sed = cmd "sed"
-
-
-awk :: CmdParams p => p
-awk = cmd "awk"
-
-
-cat' :: CmdParams p => p
-cat' = cmd "cat"
-
+reachVersion :: App
+reachVersion =
+  let v = T.pack V.versionStr
+  in write [N.text|
+    if [ "x$$REACH_VERSION" = "x" ]; then
```

Comment:
`$$`?

Reply:
Ah, this is Haskell templates. That's scary

Reply:
Yeah... It's pretty gnarly doing it this way.

My goal is to gradually push as much out of the quasi-quotes as possible and keep only those snippets small enough that template indirection would be overkill.